cc_plugin_ncei/ncei_base.py

```diff
@@ -0,0 +1,689 @@
+                enumerated_flag_variable:long_name = "" ; //................. RECOMMENDED - Provide a descriptive, long name for this variable. 
+                enumerated_flag_variable:flag_values = ; //.................. REQUIRED    - Provide a comma-separated list of flag values that map to the flag_meanings.
+                enumerated_flag_variable:flag_meanings = "" ; //............. REQUIRED    - Provide a space-separated list of meanings corresponding to each of the flag_values
+                enumerated_flag_variable:references = "" ; //................ RECOMMENDED - Published or web-based references that describe the data or methods used to produce it.
+                enumerated_flag_variable:comment = "" ; //................... RECOMMENDED - Add useful, additional information here.
+                '''
+        #Check the qaqc variables to ensure they are good
+        results = []
+        msgs = []
+        std_name_list = [getattr(dataset.variables[var], 'standard_name', None) for var in dataset.variables]
+        for name in dataset.variables:
+            if hasattr(dataset.variables[name],'flag_meanings'):
+                #Check Standard Name
+                std_check = 0
+                msgs = []
+
+                std_name = getattr(dataset.variables[name], 'standard_name', 'thereis noname')
+                std_name_split = std_name.split(' ')
+                if std_name_split[0] in std_name_list:
+                    std_check = std_check + 1
+                else:
+                    msgs.append('Standard Name does not reference a variable in the dataset.')
+                if std_name_split[1] == 'status_flag':
+                    std_check = std_check + 1
+                else:
+                    msgs.append('Standard Name does not end in "status_flag"')
+                results.append(Result(BaseCheck.HIGH, (std_check,2), (name,'standard_name'), msgs))
+
+                #Check Flag Mask / Flag Values
+                msgs = []
+                mask_val_check = False
+                test_name = 'flag_masks_values'
+                if 'bool' in str(dataset.variables[name].dtype):
+                    mask_val_check = hasattr(dataset.variables[name],'flag_masks')
+                    test_name = 'flag_masks'
+                    mask_length = len(getattr(dataset.variables[name], 'flag_masks', None)) #Used in length check
+                elif 'int' in str(dataset.variables[name].dtype):
+                    mask_val_check = hasattr(dataset.variables[name], 'flag_values')
+                    test_name = 'flag_values'
+                    mask_length = len(getattr(dataset.variables[name], 'flag_values', None)) #Used in length check
```

Comment:
The length of the 'flag_values' should be computed as 

```python
flag_values = getattr(dataset.variables[name], 'flag_values', None)
if flag_values:
    if isinstance(flag_values, basestring):  # Make sure Kyle's workaround is somewhere for py3
        flag_values = flag_values.split(' '):
    # Otherwise they are a numpy array
mask_length = len(flag_values)
```